src/assembler/compiled.rs

```diff
@@ -81,15 +81,19 @@ impl Assembler for CompiledAssembler {
 
             match res {
                 shared::SpirvResult::Success => {
-                    if text.is_null() && !options.print {
-                        return Err(crate::error::Error {
+                    if text.is_null() {
+                        if !options.print {
```

Comment:
I think this line make it strange to report an error while printing, should we accept null text as empty string for all success case?

Reply:
if print option is off, then text should not be null and error should be reported.

Reply:
This line here will not report error while printing, it is ``!options.print``

Reply:
Whatever print or not, if `ret` is Success, there should be no error.